scss/forms/_form-check.scss

```diff
@@ -107,6 +107,46 @@
 .form-switch {
   padding-left: $form-switch-padding-start;
 
+  @each $state, $value in $theme-colors {
+    &.switch-on-#{$state} {
+      .form-check-input {
+        &:checked {
+          background-color: $value;
+          border-color: $value;
+          &:focus,
+          &:active {
+            border-color: $value;
+            outline: 0;
+            box-shadow: 0 0 $input-btn-focus-blur $input-btn-focus-width rgba($value, $input-btn-focus-color-opacity);
+          }
+        }
+      }
+    }
+
+    &.switch-off-#{$state} {
+      &:not([class*="switch-on-"]) .form-check-input:checked {
+        background-color: $white;
+        background-image: escape-svg($form-switch-bg-image);
+        border-color: $form-switch-color;
+      }
+
+      .form-check-input {
+        &:not(:checked) {
+          background-color: $value;
+          background-image: escape-svg($form-switch-bg-off-image);
+          border-color: $value;
+
+          &:focus,
+          &:active {
+            border-color: $value;
+            outline: 0;
+            box-shadow: 0 0 $input-btn-focus-blur $input-btn-focus-width rgba($value, $input-btn-focus-color-opacity);
+          }
+        }
+      }
```

Comment:
This would probably need some refactor if we follow my first suggestions, obviously :D

Reply:
yep, no problem :-)